flaml/searcher/search_thread.py

```diff
@@ -70,7 +71,7 @@ def _update_speed(self):
         # calculate speed; use 0 for invalid speed temporarily
         if self.obj_best2 > self.obj_best1: 
             self.speed = (self.obj_best2 - self.obj_best1) / (
-                self.cost_total - self.cost_best2)
+                self.cost_total - self.cost_best2 + self.eps)
```

Comment:
why adding self.eps? (presumably self.cost_total should not equal to self.cost_best2)

Reply:
test_nested fails in test_tune.py with DivideByZeroException

Reply:
why does it fail?

Reply:
self.cost_total = self.cost_best2

Reply:
So we have self.obj_best2 > self.obj_best1 and  self.cost_total = self.cost_best2 at the same time?

Reply:
yes